app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/wiki/selector/WikiSelectorScreen.kt

```diff
@@ -0,0 +1,217 @@
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyItemScope
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.material.Divider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.rememberPagerState
+import io.eugenethedev.taigamobile.R
+import io.eugenethedev.taigamobile.ui.components.appbars.ClickableAppBar
+import io.eugenethedev.taigamobile.ui.components.buttons.PlusButton
+import io.eugenethedev.taigamobile.ui.components.containers.ContainerBox
+import io.eugenethedev.taigamobile.ui.components.containers.HorizontalTabbedPager
+import io.eugenethedev.taigamobile.ui.components.containers.Tab
+import io.eugenethedev.taigamobile.ui.components.dialogs.EmptyWikiDialog
+import io.eugenethedev.taigamobile.ui.components.loaders.CircularLoader
+import io.eugenethedev.taigamobile.ui.screens.main.Routes
+import io.eugenethedev.taigamobile.ui.utils.LoadingResult
+import io.eugenethedev.taigamobile.ui.utils.navigateToWikiPageScreen
+import io.eugenethedev.taigamobile.ui.utils.subscribeOnError
+
+@Composable
+fun WikiSelectorScreen(
+    navController: NavController,
+    showMessage: (message: Int) -> Unit = {},
+) {
+    val viewModel: WikiSelectorViewModel = viewModel()
+
+    val projectName by viewModel.projectName.collectAsState()
+
+    val wikiLinks by viewModel.wikiLinksResult.collectAsState()
+    wikiLinks.subscribeOnError(showMessage)
+
+    val wikiPages by viewModel.wikiPagesResult.collectAsState()
+    wikiPages.subscribeOnError(showMessage)
+
+    LaunchedEffect(Unit) {
+        viewModel.getWikiPage()
+        viewModel.getWikiLinks()
+    }
+
+    if (wikiLinks is LoadingResult || wikiPages is LoadingResult) {
+        Box(
+            modifier = Modifier.fillMaxSize(),
+            contentAlignment = Alignment.Center
+        ) {
+            CircularLoader()
+        }
+    } else {
+        val wikiPagesSlug = wikiPages.data.orEmpty().map { it.slug }
+        val wikiLinksSlug = wikiLinks.data.orEmpty().filter { it.ref !in wikiPagesSlug }.map { it.ref }
+
+        WikiPageSelectorContent(
+            projectName = projectName,
+            bookmarksTitles = wikiLinks.data.orEmpty().map { it.title },
```

Comment:
It's better to use consistent names - e.g `bookmarks` and `allPages`. Also, as I said earlier, it's not enought to just store the titles, you will need slugs to navigate anyway. So it's better to use something like map of slugs with their respective titles